src/bulletin/bulletin.service.ts

```diff
@@ -69,4 +69,98 @@ export class BulletinService {
       }
     });
   }
+
+  async updateBulletin(
+    id: number,
+    updateBulletinDto: BulletinDTO,
+    pdfAttachments: Express.Multer.File[],
+  ) {
+    return this.prismaService.$transaction(async (tx) => {
+      try {
+        const updatedBulletin = await tx.bulletin.update({
+          where: { id },
+          data: {
+            category_id: updateBulletinDto.category_id,
+            title: updateBulletinDto.title,
+            content: updateBulletinDto.content,
+            author: updateBulletinDto.author,
+          },
```

Comment:
Use spread operator. No need to specify each attribute here.

`...updateBulletinDto`

Reply:
No, I rather not do that, updatebulletinDto is a request payload from frontend and it does not have the id to define the pdf. ID is only define by the http params.